vars/dockerExecuteOnKubernetes.groovy

```diff
@@ -162,8 +160,10 @@ void executeOnPod(Map config, utils, Closure body) {
      * In case third case, we need to create the 'container' stash to bring the modified content back to the host.
      */
     try {
-        if (config.containerName && config.stashContent.isEmpty()){
-            config.stashContent.add(stashWorkspace(config, 'workspace'))
```

Comment:
It was on purpose (performance) to only stash complete workspace when no stashContent was  provided. Proposed change is incompatible to this assumption.

Reply:
I agree though to add a unique id to the name of the stash instead of using a fixed value.

Reply:
The current behaviour causes a failure since we rely on a stash which was created too early. Maybe we should invalidate stashes when we return from a execution performed via docker/kubernetes